analyzers/src/SonarAnalyzer.CSharp/Rules/MethodOverloadsShouldBeGrouped.cs

```diff
@@ -51,28 +61,13 @@ protected override MemberInfo CreateMemberInfo(SyntaxNodeAnalysisContext c, Memb
             return null;
         }
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
-            context.RegisterSyntaxNodeActionInNonGenerated(c =>
-            {
-                var typeDeclaration = (TypeDeclarationSyntax)c.Node;
-                CheckMembers(c, typeDeclaration.Members);
-            },
-            SyntaxKind.ClassDeclaration,
-            SyntaxKind.InterfaceDeclaration,
-            SyntaxKind.StructDeclaration);
-        }
-
-        private static bool IsValidMemberForOverload(MemberDeclarationSyntax member)
-        {
-            if (member is MethodDeclarationSyntax methodDeclaration)
-            {
-                return methodDeclaration.ExplicitInterfaceSpecifier == null;
-            }
-            return true;
-        }
+        protected override IEnumerable<MemberDeclarationSyntax> GetMemberDeclarations(SyntaxNode node) =>
+            ((TypeDeclarationSyntax)node).Members;
 
-        private static bool IsStatic(BaseMethodDeclarationSyntax declaration) => declaration.Modifiers.Any(x => x.Kind() == SyntaxKind.StaticKeyword);
+        private static bool IsValidMemberForOverload(MemberDeclarationSyntax member) =>
+            !(member is MethodDeclarationSyntax methodDeclaration) || methodDeclaration.ExplicitInterfaceSpecifier == null;
```

Comment:
I think this is a bit more concise and easier to read:
```suggestion
            (member as MethodDeclarationSyntax)?.ExplicitInterfaceSpecifier == null;
```